antelope_contracts/tests/erc20/integrated_tests.cpp

```diff
@@ -0,0 +1,482 @@
+    push_action(evmtok_account, "setingressfee"_n, evmtok_account, 
+        mvo()("token_contract", token_account)("ingress_fee", make_asset(0, token_symbol)));
+
+    transfer_token(token_account, "alice"_n, evmtok_account, make_asset(10000000, token_symbol), evm1.address_0x().c_str());
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 3000000000); // +1000000000
+    BOOST_REQUIRE(69999700 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount()); // 
+    BOOST_REQUIRE(30000300 == get_balance(evmtok_account, token_account, symbol::from_string("4,USDT")).get_amount()); 
+    produce_block();
+
+    BOOST_REQUIRE_EXCEPTION(push_action(evmtok_account, "setingressfee"_n, evmtok_account, 
+        mvo()("token_contract", token_account)("ingress_fee", make_asset(0, symbol::from_string("4,USDC"))));,
+            eosio_assert_message_exception, eosio_assert_message_is("token not registered"));
+
+    BOOST_REQUIRE_EXCEPTION(push_action(evmtok_account, "setingressfee"_n, evmtok_account, 
+        mvo()("token_contract", token_account)("ingress_fee", make_asset(0, symbol::from_string("2,USDT"))));,
+            eosio_assert_message_exception, eosio_assert_message_is("incorrect precision for registered token"));
+
+}
+FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(it_set_egress_fee, it_tester)
+try {
+    constexpr intx::uint256 minimum_natively_representable = intx::exp(10_u256, intx::uint256(18 - 4));
+
+    BOOST_REQUIRE_EXCEPTION(push_action(evmtok_account, "setegressfee"_n, evmtok_account, 
+        mvo()("token_contract", token_account)("token_symbol_code", "USDT")("egress_fee", make_asset(50))),
+        eosio_assert_message_exception, eosio_assert_message_is("egress fee must be at least as large as the receiver's minimum fee"));
+
+    produce_block();
+
+    BOOST_REQUIRE(100 * minimum_natively_representable == egressFee()); // was 0.01
+
+    produce_block();
+    // set to 0.5
+    push_action(evmtok_account, "setegressfee"_n, evmtok_account, 
+        mvo()("token_contract", token_account)("token_symbol_code", "USDT")("egress_fee", make_asset(5000)));
+    
+    BOOST_REQUIRE(5000 * minimum_natively_representable == egressFee());
+
+    produce_block();
+   
+    BOOST_REQUIRE_EXCEPTION(push_action(evmtok_account, "setegressfee"_n, evm_account, 
+        mvo()("token_contract", token_account)("token_symbol_code", "USDT")("egress_fee", make_asset(1000))),
+        missing_auth_exception, eosio::testing::fc_exception_message_starts_with("missing authority of eosio.evmtok"));
+
+
+    produce_block();
+    auto delegated_auth = authority( 1, {},
+                          {
+                            { .permission = {erc20_account,config::active_name}, .weight = 1}
+                          });
+
+    set_authority( erc20_account, config::active_name,  delegated_auth );
+
+    produce_block();
+
+    BOOST_REQUIRE_EXCEPTION(push_action(evmtok_account, "setegressfee"_n, evmtok_account, 
```

Comment:
This part of the test with changing the authority is again not necessary and is a distraction to be removed after you bring in the latest changes that removes `evmtok_account`.

Reply:
done